### DISCUSSION TITLE
Is there a way to locate the dune file which defines a certain library

### DISCUSSION THREAD

**Discussion by haochenx** [ocaml/dune#7036] [Q&A] [answered]
Since with dune there is a great degree of freedom about how to organize the source code, there are many times where I found it difficult where a certain library is defined.

Is there a cli command, probably something like `dune locate library foo`, which would output the path of the dune file where the library stanza is defined? This would be extremely helpful for browsing unfamiliar codebase as well I believe.

**Comment by Alizter** [comment, marked as answer]
There is the `dune describe` command which is `dune describe workspace`. This will print information about the workspace. In there you will find libraries and their source directories. You can grep the output for the library you want.

**Reply by haochenx** [reply]
Indeed! The only problem is that the output is not super "grep-friendly"..

Of course we can use the `-C` option to display lines around the target line, but it's not perfect.
(it would be nice to have a sexp specific grepping that can output the top-level sexp of the matches....)
